src/ORM/DataObject.php

```diff
@@ -3175,14 +3175,24 @@ public static function reset()
     /**
      * Return the given element, searching by ID
      *
-     * @param string $callerClass The class of the object to be returned
+     * @param string $callerClass The class of the object to be returned (optional if called from extending class)
      * @param int $id The id of the element
      * @param boolean $cache See {@link get_one()}
      *
      * @return DataObject The element
      */
-    public static function get_by_id($callerClass, $id, $cache = true)
+    public static function get_by_id($callerClass, $id = null, $cache = true)
     {
+        if(is_numeric($callerClass)) {
+            // Alternate signature to allow calling as MyDataObject::get_by_id($id)
+            $cache = is_null($id) ? true : !!$id;
+            $id = $callerClass;
+            $callerClass = get_called_class();
```

Comment:
Best practice is create new variables instead of re-assigning method argument variables.